Sources/Honeycomb/HoneycombNavigationSpanInstrumentation.swift

```diff
@@ -0,0 +1,90 @@
+import OpenTelemetryApi
+import OpenTelemetrySdk
+import SwiftUI
+
+private let NavigationInstrumentationName = "@honeycombio/instrumentation-navigation"
+private let NavigationBaggageKey = EntryKey(name: "CurrentNavigationPath")!
+private let NavigationSpanName = "Navigation"
+private let UnencodablePath = "<unencodable path>"
+
+private func activeBaggage() -> Baggage? {
+    return OpenTelemetry.instance.contextProvider.activeBaggage
+}
+
+func getTracer() -> Tracer {
+    return OpenTelemetry.instance.tracerProvider.get(
+        instrumentationName: NavigationInstrumentationName,
+        instrumentationVersion: honeycombLibraryVersion
+    )
+}
+
+@available(iOS 16.0, macOS 12.0, *)
+func reportNavigation(path: NavigationPath) {
+    if let codablePath = path.codable {
+        reportNavigation(path: codablePath)
+    } else {
+        reportNavigation(path: UnencodablePath)
+    }
+
+}
+
+var currentNavigationPath: String? = nil
+
+func reportNavigation(path: Encodable) {
+    do {
+        let encoder = JSONEncoder()
+        let data = try encoder.encode(path)
+        let pathStr = String(decoding: data, as: UTF8.self)
+        print("current path: \(pathStr)")
+
+        currentNavigationPath = pathStr
```

Comment:
Ok this looks super dumb.

BUT! The "correct" way to do this (with baggage or context) seems busted. 
- otel swift doesn't seem to expose an open "write whatever you want into context" method (and if it does, i couldn't find it)
- otel swift DOES offer a way to dumb arbitrary data into baggage, but baggage is immutable, meaning every new value in baggage needs to also call setActiveBaggage(), meaning that some other code could come along and randomly decide to nuke it without us being able to do anything about it.

ie. instead of just assigning to this global variable, I was doing this:

```swift
let baggage = OpenTelemetry.instance.baggageManager.baggageBuilder()
    .put(key: EntryKey("CurrentNavigationPath")!, value: EntryValue(string: pathStr)!, metadata: nil)
    .build()

OpenTelemetry.instance.contextProvider.setActiveBaggage(baggage)
```

Then I was relying on our BaggageSpanProcessor to pull the value out of baggage and stick it on each span's attributes. However, this worked for some spans and then for others, they just wouldn't get the right attribute. As far as I could tell, something in the process was clearing out the active Baggage, but I couldn't figure out what was doing it.

Reply:
Is this something we should fix in upstream Otel? I agree it's annoying that baggage isn't mutable, and even if it were immutable, it seems like there should be an API to extend it with a new entry.

We could write a Baggage extension that copies the existing keys and adds new ones, and then work on upstreaming it later.

Reply:
The problem isn't setting Baggage, it's convincing everything to use the same extension/methods. 

I was able to create new baggage with what I needed, but some other code somewhere else seemed to overwriting it :sob:

Reply:
circling back: based on internal discussion on slack, it sounds like baggage isn't even the best choice for this pattern. 

Right now I'm just using a global mutable variable, which feels silly and unconventional. Is there a better Swift-y conventional API that I should be using instead?

Reply:
I think having a mutable global singleton is probably the best option, since that's how the OpenTelemetry SDK already exposes its APIs, and this is truly global state.

I wonder if we can solve both this and the naming problem with the same solution. Maybe if we have a singleton like `HoneycombNavigation` and call `HoneycombNavigation.setCurrentScreen()`, then we could record those as navigation events as well as keeping track of the state for a custom span processor to add the attributes?

I guess that's not any better than `Honeycomb.setCurrentScreen()` though. Well, I'm open to suggestions. But I kinda like the idea of having a straightforward "This is the screen you're on" API that logs the event (for Timeline Analysis) and also sets the global attribute. Then that API could work with _any_ UI framework, and we should even call it for the UIKit instrumentation, assuming they aren't constantly clobbering each other.

Reply:
I think I misread your previous suggestion about putting it on the global honeycomb object. I think that works best. 

I'll rewrite this into a shared singleton and then expose that from the main honeycomb object.